core/themes/bartik/templates/node.tpl.php

```diff
@@ -119,7 +117,7 @@
     </footer>
   <?php endif; ?>
 
-  <?php if ($page && isset($comments['comments'])): ?>
+  <?php if ($page && !empty($comments)): ?>
```

Comment:
this change will detect both the comments and the comment form. Is that what was intended?  Previously we were just looking for comments.